src/workspace/internal/workspacesettingsstream.cpp

```diff
@@ -78,7 +78,7 @@ void WorkspaceSettingsStream::write(const AbstractDataPtrList& settingsList, IOD
 void WorkspaceSettingsStream::writeSettings(XmlWriter& writer, const AbstractDataPtr& data) const
 {
     SettingsDataPtr settings = std::dynamic_pointer_cast<SettingsData>(data);
-    IF_ASSERT_FAILED(settings) {
+    if (!settings) {
```

Comment:
Can you please explain the reason for this change?

Reply:
because I created stubs for all data and we need to write them to a file for recovery after restart

Reply:
Seems there is a bug in NewWorkspaceData::addStubData. You should add an object of a concrete type. At the moment you add only objects of UiArrangementData type

Reply:
reverted, fixed bug